great_expectations/util.py

```diff
@@ -1214,55 +1215,19 @@ def filter_properties_dict(
     return properties
 
 
-@overload
-def deep_filter_properties_iterable(
-    properties: dict,
-    keep_fields: Optional[Set[str]] = ...,
-    delete_fields: Optional[Set[str]] = ...,
-    clean_nulls: bool = ...,
-    clean_falsy: bool = ...,
-    keep_falsy_numerics: bool = ...,
-    inplace: bool = ...,
-) -> dict:
-    ...
-
-
-@overload
-def deep_filter_properties_iterable(
-    properties: list,
-    keep_fields: Optional[Set[str]] = ...,
-    delete_fields: Optional[Set[str]] = ...,
-    clean_nulls: bool = ...,
-    clean_falsy: bool = ...,
-    keep_falsy_numerics: bool = ...,
-    inplace: bool = ...,
-) -> list:
-    ...
-
-
-@overload
-def deep_filter_properties_iterable(
-    properties: set,
-    keep_fields: Optional[Set[str]] = ...,
-    delete_fields: Optional[Set[str]] = ...,
-    clean_nulls: bool = ...,
-    clean_falsy: bool = ...,
-    keep_falsy_numerics: bool = ...,
-    inplace: bool = ...,
-) -> set:
-    ...
+T = TypeVar("T", bound=Union[dict, list, tuple, set])
```

Comment:
This change leads to typing errors when assigning to `properties` in the body of the function.
A bit confused as to why. I would have expected the type-narrowing to convince the checker that `properties` is either a `dict`, `tuple`, etc. when the assignment is performed.